poc_mobile_verifier/src/heartbeats.rs

```diff
@@ -43,75 +55,19 @@ impl Heartbeat {
 impl From<Share> for Heartbeat {
     fn from(share: Share) -> Self {
         Self {
-            id: share.pub_key,
+            hotspot_key: share.pub_key,
+            cbsd_id: share.cbsd_id,
             reward_weight: share.reward_weight,
             timestamp: share.timestamp,
         }
     }
 }
 
-#[derive(Clone)]
-pub struct HeartbeatValue {
-    pub reward_weight: Decimal,
-    pub timestamp: NaiveDateTime,
-}
-
-#[derive(Default, Clone)]
-pub struct Heartbeats {
-    pub heartbeats: HashMap<PublicKey, HeartbeatValue>,
-}
-
-impl Heartbeats {
-    /// Constructs a new heartbeats collection, starting by pulling every heartbeat
-    /// since the end of the last rewardable period (`starting`).
-    pub async fn validated(
-        exec: impl sqlx::Executor<'_, Database = sqlx::Postgres> + Copy,
-        starting: DateTime<Utc>,
-    ) -> std::result::Result<Self, sqlx::Error> {
-        let mut heartbeats = HashMap::new();
-        let mut rows =
-            sqlx::query_as::<_, Heartbeat>("SELECT * FROM heartbeats WHERE timestamp >= $1")
-                .bind(starting)
-                .fetch(exec);
-
-        while let Some(Heartbeat {
-            id,
-            reward_weight,
-            timestamp,
-        }) = rows.try_next().await?
-        {
-            heartbeats.insert(
-                id,
-                HeartbeatValue {
-                    reward_weight,
-                    timestamp,
-                },
-            );
-        }
-        Ok(Self { heartbeats })
-    }
-}
-
-impl FromIterator<Share> for Heartbeats {
+impl FromIterator<Share> for Vec<Heartbeat> {
     fn from_iter<T>(iter: T) -> Self
     where
         T: IntoIterator<Item = Share>,
     {
-        let mut heartbeats = HashMap::default();
-        for share in iter.into_iter() {
-            let Heartbeat {
-                id,
-                reward_weight,
-                timestamp,
-            } = Heartbeat::from(share);
-            heartbeats.insert(
-                id,
-                HeartbeatValue {
-                    reward_weight,
-                    timestamp,
-                },
-            );
-        }
-        Self { heartbeats }
+        iter.into_iter().map(Heartbeat::from).collect()
```

Comment:
Since collect takes a FromIterator bound, you might as well entirely remove this impl and put this line wherever you want heartbeats

Reply:
Fixed 7dc32cc